### DISCUSSION TITLE
Just curious Hangfire or Quartz

### DISCUSSION THREAD

**Discussion by bytesfromali** [elsa-workflows/elsa-core#1040] [General]
Planning to design a workflow product on top of Elsa, just curious which one to choose Hangfire/ Quartz.

Is is there anything in general which is supported by one and not the other, because I have used them in normal Web Apps for background jobs interchangeably.

Asking this to understand the direction of the two packages, so that I can make a sound decision as well as contribute in future.

**Comment by sfmskywalker** [comment]
Honestly, I don't know either.

Both are equally capable of running scheduled work, although I have found Quartz to be more accurate than Hangfire in terms of running a scheduled job at a specific time, while Hangfire might be off with a few seconds. This also depends on the configured schedule polling interval setting.

One consideration to make when using one or the other is if for example you already are using Hangfire in your project, it might make sense to use Hangfire as the provider for Elsa temporal activities as well. But if for some reason you like Quartz better because it's more accurate, then it's perfectly fine to use it together with Hangfire.

Ultimately they are both here to give you a choice. Technically, they both work great and are here to stay.

**Reply by bytesfromali** [reply]
Thanks, You rightly pointed out about making decision on what is already being used. Surely gonna consider on that basis.
